src/main/java/movie/storage/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,46 @@
+package movie.storage.dao.impl;
+
+import java.util.Optional;
+import movie.storage.dao.UserDao;
+import movie.storage.exception.IncorrectDataException;
+import movie.storage.lib.Dao;
+import movie.storage.model.User;
+import movie.storage.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class UserDaoImpl implements UserDao {
+    @Override
+    public User add(User user) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(user);
+            transaction.commit();
+            return user;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new IncorrectDataException("Unable to add user" + user, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<User> findByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("from User where email= :email", User.class)
+                    .setParameter("email", email)
+                    .uniqueResultOptional();
+        } catch (Exception e) {
+            throw new IncorrectDataException("Unable to find user with email - " + email, e);
```

Comment:
this exception won't suit here (if email is incorrect you will return optional empty, you may get exception here if query is wrong etc, but not email) So remove catch block at all, or make it informative (first option will preferable)

Reply:
Done.